src/watson/models.py

```diff
@@ -70,14 +72,21 @@ class SearchEntry(models.Model):
 
     meta_encoded = models.TextField()
 
-    @property
+    def _deserialize_meta(self):
+        from watson.registration import SearchEngine
+        engine = SearchEngine._created_engines[self.engine_slug]
+        model = self.content_type.model_class()
```

Comment:
perhaps we could have another `get_adapter`-like method that took `content_type_id` as an argument?